data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java

```diff
@@ -86,6 +72,10 @@ public Certificate getACMCertificate(final String acmArn, final String passphras
         }
     }
 
+    private String getAcmPrivateKeyPassword(final OTelTraceSourceConfig oTelTraceSourceConfig) {
+        return Optional.ofNullable(oTelTraceSourceConfig.getAcmPrivateKeyPassword()).orElse(UUID.randomUUID().toString());
```

Comment:
The private key from ACM is encrypted. It's privateKey password that will be used to decrypt the private key. It can be configured and can be later be used to decrypt the private key manually using openssl commands or else using UUID to generate the password. In later case the password is in memory and can't be used for any manual inspection/generation of private key.